Assets/Scripts/UI/InGameUI/MenuTop/CurrencyDisplay.cs

```diff
@@ -0,0 +1,34 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System.Text;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class CurrencyDisplay : MonoBehaviour
+{
+    public Text text;
+    
+    // Update is called once per frame
+    private void Update()
+    {
+        string[] currencies = World.Current.Wallet.GetCurrencyNames();
```

Comment:
You can cache this variable outside of Update and get it during Start, we don't need to get this every Update().